sw/device/lib/testing/csrng_testutils.c

```diff
@@ -14,64 +14,67 @@ enum {
   kNumOutputWordsMax = 16,
 };
 
-void csrng_testutils_cmd_ready_wait(const dif_csrng_t *csrng) {
+status_t csrng_testutils_cmd_ready_wait(const dif_csrng_t *csrng) {
   dif_csrng_cmd_status_t cmd_status;
+  status_t res;
   do {
-    CHECK_DIF_OK(dif_csrng_get_cmd_interface_status(csrng, &cmd_status));
-    CHECK(cmd_status.kind != kDifCsrngCmdStatusError);
+    res.value = TRY(dif_csrng_get_cmd_interface_status(csrng, &cmd_status));
+    ASSERT_TRUE(cmd_status.kind != kDifCsrngCmdStatusError);
   } while (cmd_status.kind != kDifCsrngCmdStatusReady);
+  return res;
 }
 
-void csrng_testutils_cmd_generate_run(const dif_csrng_t *csrng,
-                                      uint32_t *output, size_t output_len) {
-  csrng_testutils_cmd_ready_wait(csrng);
-  CHECK_DIF_OK(dif_csrng_generate_start(csrng, output_len));
+status_t csrng_testutils_cmd_generate_run(const dif_csrng_t *csrng,
+                                          uint32_t *output, size_t output_len) {
+  status_t res = csrng_testutils_cmd_ready_wait(csrng);
```

Comment:
Is this `res` used? I see its value is overwritten on line 37, are we only using this `res` for its error?

Reply:
Fixed